### PR TITLE
Bump prometheus version to v2.36.2

### PR DESCRIPTION
**Pull request by zhouke1991**
## What this PR does / why we need it

Bump prometheus and other related components to a new version.
## Which issue(s) this PR fixes

Bump prometheus to v2.36.2.
Bump prometheus-node-exporter to v1.3.1.
Bump prometheus-alertmanager to v0.24.0.
Bump prometheus-pushgateway to v1.4.2.
Bump prometheus-configmap-reload to v0.7.1.
Bump kube-state-metrics to v.2.5.0.

In addition, cadvisor is going to be dropped in the new version, because it is not used by Prometheus, what works is the cadvisor in kubelet. And it is also not in helm charts of [prometheus community](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus/templates)

### CODE REVIEW

**Comment by hhhybb** [comment]
@seemiller  Could you please review this Prometheus bump PR? Thanks.

File: addons/packages/prometheus/metadata.yaml [hunk 1]
```diff
@@ -9,9 +9,7 @@ spec:
   shortDescription: "A time series database for your metrics"
   providerName: VMware
   maintainers:
-    - name: Anil Kodali
-    - name: John McBride
```
Comment (jpmcb) [inline addons/packages/prometheus/metadata.yaml:13]: 👍🏼 - thanks for updating this!

**Review by jpmcb — APPROVED** [review]
Hi @hhhybb - thanks for opening this.

That Jira link is not accessible outside of Vmware engineering: can you replace it with a quick description of what this PR does? Otherwise this looks good!

I see you're in the vmware GitHub org; I can go ahead and add you to the @vmware-tanzu/pkg-prometheus-owners team so you get pinged when that team is mentioned?

**Comment by hhhybb** [comment]
@jpmcb Thanks for your updates. I have updated the description and it is an honor to join the pkg-prometheus-owner team. Thanks.

**Pull request merged**